ansible/roles/cli/tasks/deploy.yml

```diff
@@ -3,26 +3,19 @@
 
 - name: "ensure nginx directory for cli exists"
   file:
-    path: "{{ cli_nginx_dir  }}"
+    path: "{{ cli_nginx_dir }}"
     state: directory
 
-- include: docker_login.yml
+- set_fact:
+    cli_installation_mode="{{ openwhisk_cli.installation_mode }}"
 
-- name: "pull the {{ docker_image_tag }} image of cli"
-  shell: "docker pull {{ docker_registry }}{{ docker_image_prefix }}/cli:{{ docker_image_tag }}"
-  when: docker_registry != ""
+- name: "download the binaries from the release page of OpenWhisk CLI"
+  include: download_openwhisk_cli.yml
+  with_items: "{{ cli_os }}"
+  when: cli_installation_mode == "download"
 
-- name: "run Docker"
-  shell: "docker run --name cli {{ docker_registry }}{{ docker_image_prefix }}/cli:{{ docker_image_tag }}"
-
-- name: "copy all the cli binaries into nginx"
-  shell: "docker cp cli:/src/github.com/go-whisk-cli/build/. '{{ cli_nginx_dir }}'"
-  ignore_errors: yes
-
-- name: "stop Docker"
-  shell: "docker stop cli"
-
-- name: "remove cli container after publishing"
-  shell: "docker rm cli"
+- name: "download the source code if necessary and build the binaries locally"
```

Comment:
openwhisk_cli:
  installation_mode: download
  tag: latest
  url: https://github.com/openwhisk/openwhisk-cli.git
  location: "{{ openwhisk_home }}/../openwhisk-cli"
  version: "HEAD"
  repo_update: "no"

This is option for user to use.
The user can specify installation_mode to download or build. If it is set to download, openwhisk cli binaries will be downloaded. If it is set to build, a local dir containing source code will be provided, which is location. If the location is not available locally, openwhisk cli repo will be downloaded into this location. If it already exists and repo_update is set to no, we do not download the source code of openwhisk cli repo and build the binaries locally directly.